{
  "cells": [
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "qw4ZPLgEZoOA",
        "outputId": "4d07160d-d693-4bdb-ced6-42da9a913f27"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\r0% [Working]\r            \rGet:1 http://security.ubuntu.com/ubuntu jammy-security InRelease [129 kB]\n",
            "\r0% [Waiting for headers] [1 InRelease 0 B/129 kB 0%] [Waiting for headers] [Con\r                                                                               \rHit:2 http://archive.ubuntu.com/ubuntu jammy InRelease\n",
            "\r0% [1 InRelease 14.2 kB/129 kB 11%] [Waiting for headers] [Connected to r2u.sta\r                                                                               \rGet:3 https://cloud.r-project.org/bin/linux/ubuntu jammy-cran40/ InRelease [3,632 B]\n",
            "\r                                                                               \rHit:4 https://cli.github.com/packages stable InRelease\n",
            "\r0% [Waiting for headers] [1 InRelease 14.2 kB/129 kB 11%] [Connected to r2u.sta\r                                                                               \rGet:5 http://archive.ubuntu.com/ubuntu jammy-updates InRelease [128 kB]\n",
            "Get:6 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  InRelease [1,581 B]\n",
            "Get:7 https://r2u.stat.illinois.edu/ubuntu jammy InRelease [6,555 B]\n",
            "Get:8 http://archive.ubuntu.com/ubuntu jammy-backports InRelease [127 kB]\n",
            "Get:9 https://cloud.r-project.org/bin/linux/ubuntu jammy-cran40/ Packages [83.6 kB]\n",
            "Get:10 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  Packages [2,153 kB]\n",
            "Get:11 http://security.ubuntu.com/ubuntu jammy-security/restricted amd64 Packages [6,008 kB]\n",
            "Get:12 https://ppa.launchpadcontent.net/deadsnakes/ppa/ubuntu jammy InRelease [18.1 kB]\n",
            "Get:13 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 Packages [1,592 kB]\n",
            "Get:14 http://security.ubuntu.com/ubuntu jammy-security/universe amd64 Packages [1,286 kB]\n",
            "Hit:15 https://ppa.launchpadcontent.net/graphics-drivers/ppa/ubuntu jammy InRelease\n",
            "Get:16 https://r2u.stat.illinois.edu/ubuntu jammy/main amd64 Packages [2,836 kB]\n",
            "Get:17 http://security.ubuntu.com/ubuntu jammy-security/main amd64 Packages [3,535 kB]\n",
            "Get:18 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 Packages [3,870 kB]\n",
            "Get:19 http://archive.ubuntu.com/ubuntu jammy-updates/restricted amd64 Packages [6,222 kB]\n",
            "Hit:20 https://ppa.launchpadcontent.net/ubuntugis/ppa/ubuntu jammy InRelease\n",
            "Get:21 https://r2u.stat.illinois.edu/ubuntu jammy/main all Packages [9,491 kB]\n",
            "Get:22 https://ppa.launchpadcontent.net/deadsnakes/ppa/ubuntu jammy/main amd64 Packages [38.5 kB]\n",
            "Fetched 37.5 MB in 2s (17.4 MB/s)\n",
            "Reading package lists... Done\n",
            "W: Skipping acquire of configured file 'main/source/Sources' as repository 'https://r2u.stat.illinois.edu/ubuntu jammy InRelease' does not seem to provide it (sources.list entry misspelt?)\n",
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "samtools is already the newest version (1.13-4).\n",
            "Suggested packages:\n",
            "  python3-numpy python3-matplotlib texlive-latex-recommended\n",
            "The following NEW packages will be installed:\n",
            "  bcftools bedtools bwa\n",
            "0 upgraded, 3 newly installed, 0 to remove and 57 not upgraded.\n",
            "Need to get 1,455 kB of archives.\n",
            "After this operation, 4,375 kB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu jammy/universe amd64 bcftools amd64 1.13-1 [697 kB]\n",
            "Get:2 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 bedtools amd64 2.30.0+dfsg-2ubuntu0.1 [563 kB]\n",
            "Get:3 http://archive.ubuntu.com/ubuntu jammy/universe amd64 bwa amd64 0.7.17-6 [195 kB]\n",
            "Fetched 1,455 kB in 1s (1,145 kB/s)\n",
            "Selecting previously unselected package bcftools.\n",
            "(Reading database ... 121812 files and directories currently installed.)\n",
            "Preparing to unpack .../bcftools_1.13-1_amd64.deb ...\n",
            "Unpacking bcftools (1.13-1) ...\n",
            "Selecting previously unselected package bedtools.\n",
            "Preparing to unpack .../bedtools_2.30.0+dfsg-2ubuntu0.1_amd64.deb ...\n",
            "Unpacking bedtools (2.30.0+dfsg-2ubuntu0.1) ...\n",
            "Selecting previously unselected package bwa.\n",
            "Preparing to unpack .../bwa_0.7.17-6_amd64.deb ...\n",
            "Unpacking bwa (0.7.17-6) ...\n",
            "Setting up bwa (0.7.17-6) ...\n",
            "Setting up bcftools (1.13-1) ...\n",
            "Setting up bedtools (2.30.0+dfsg-2ubuntu0.1) ...\n",
            "Processing triggers for man-db (2.10.2-1) ...\n"
          ]
        }
      ],
      "source": [
        "!apt-get update -y\n",
        "\n",
        "# Ferramentas básicas de NGS\n",
        "!apt-get install -y bwa samtools bcftools bedtools\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "S8XEWufkaMpr",
        "outputId": "e2feddd4-21a9-46b0-b5f3-19537e36a08e"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "The following additional packages will be installed:\n",
            "  libdisorder0 libfastahack0 libfml0 libseqlib2 libsmithwaterman0 libssw0\n",
            "  libtabixpp0 libvcflib1 parallel sysstat\n",
            "Suggested packages:\n",
            "  ash csh fish ksh tcsh zsh isag\n",
            "The following NEW packages will be installed:\n",
            "  freebayes libdisorder0 libfastahack0 libfml0 libseqlib2 libsmithwaterman0\n",
            "  libssw0 libtabixpp0 libvcflib1 parallel sysstat\n",
            "0 upgraded, 11 newly installed, 0 to remove and 57 not upgraded.\n",
            "Need to get 4,437 kB of archives.\n",
            "After this operation, 8,967 kB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libfml0 amd64 0.1+git20190320.b499514-1 [70.6 kB]\n",
            "Get:2 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libssw0 amd64 1.1-13 [19.1 kB]\n",
            "Get:3 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libseqlib2 amd64 1.2.0+dfsg-8build1 [175 kB]\n",
            "Get:4 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libtabixpp0 amd64 1.1.0-6 [9,998 B]\n",
            "Get:5 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libdisorder0 amd64 0.0.2+git20130809.8062ee1-4 [4,712 B]\n",
            "Get:6 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libfastahack0 amd64 1.0.0+dfsg-9 [27.0 kB]\n",
            "Get:7 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libsmithwaterman0 amd64 0.0+git20160702.2610e25-12 [37.6 kB]\n",
            "Get:8 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libvcflib1 amd64 1.0.3+dfsg-1 [169 kB]\n",
            "Get:9 http://archive.ubuntu.com/ubuntu jammy/universe amd64 freebayes amd64 1.3.6-1 [1,491 kB]\n",
            "Get:10 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 sysstat amd64 12.5.2-2ubuntu0.2 [487 kB]\n",
            "Get:11 http://archive.ubuntu.com/ubuntu jammy/universe amd64 parallel all 20210822+ds-2 [1,947 kB]\n",
            "Fetched 4,437 kB in 1s (3,386 kB/s)\n",
            "Preconfiguring packages ...\n",
            "Selecting previously unselected package libfml0:amd64.\n",
            "(Reading database ... 121937 files and directories currently installed.)\n",
            "Preparing to unpack .../00-libfml0_0.1+git20190320.b499514-1_amd64.deb ...\n",
            "Unpacking libfml0:amd64 (0.1+git20190320.b499514-1) ...\n",
            "Selecting previously unselected package libssw0:amd64.\n",
            "Preparing to unpack .../01-libssw0_1.1-13_amd64.deb ...\n",
            "Unpacking libssw0:amd64 (1.1-13) ...\n",
            "Selecting previously unselected package libseqlib2:amd64.\n",
            "Preparing to unpack .../02-libseqlib2_1.2.0+dfsg-8build1_amd64.deb ...\n",
            "Unpacking libseqlib2:amd64 (1.2.0+dfsg-8build1) ...\n",
            "Selecting previously unselected package libtabixpp0:amd64.\n",
            "Preparing to unpack .../03-libtabixpp0_1.1.0-6_amd64.deb ...\n",
            "Unpacking libtabixpp0:amd64 (1.1.0-6) ...\n",
            "Selecting previously unselected package libdisorder0:amd64.\n",
            "Preparing to unpack .../04-libdisorder0_0.0.2+git20130809.8062ee1-4_amd64.deb ...\n",
            "Unpacking libdisorder0:amd64 (0.0.2+git20130809.8062ee1-4) ...\n",
            "Selecting previously unselected package libfastahack0:amd64.\n",
            "Preparing to unpack .../05-libfastahack0_1.0.0+dfsg-9_amd64.deb ...\n",
            "Unpacking libfastahack0:amd64 (1.0.0+dfsg-9) ...\n",
            "Selecting previously unselected package libsmithwaterman0:amd64.\n",
            "Preparing to unpack .../06-libsmithwaterman0_0.0+git20160702.2610e25-12_amd64.deb ...\n",
            "Unpacking libsmithwaterman0:amd64 (0.0+git20160702.2610e25-12) ...\n",
            "Selecting previously unselected package libvcflib1:amd64.\n",
            "Preparing to unpack .../07-libvcflib1_1.0.3+dfsg-1_amd64.deb ...\n",
            "Unpacking libvcflib1:amd64 (1.0.3+dfsg-1) ...\n",
            "Selecting previously unselected package freebayes.\n",
            "Preparing to unpack .../08-freebayes_1.3.6-1_amd64.deb ...\n",
            "Unpacking freebayes (1.3.6-1) ...\n",
            "Selecting previously unselected package sysstat.\n",
            "Preparing to unpack .../09-sysstat_12.5.2-2ubuntu0.2_amd64.deb ...\n",
            "Unpacking sysstat (12.5.2-2ubuntu0.2) ...\n",
            "Selecting previously unselected package parallel.\n",
            "Preparing to unpack .../10-parallel_20210822+ds-2_all.deb ...\n",
            "Adding 'diversion of /usr/bin/parallel to /usr/bin/parallel.moreutils by parallel'\n",
            "Adding 'diversion of /usr/share/man/man1/parallel.1.gz to /usr/share/man/man1/parallel.moreutils.1.gz by parallel'\n",
            "Unpacking parallel (20210822+ds-2) ...\n",
            "Setting up libdisorder0:amd64 (0.0.2+git20130809.8062ee1-4) ...\n",
            "Setting up libsmithwaterman0:amd64 (0.0+git20160702.2610e25-12) ...\n",
            "Setting up libfml0:amd64 (0.1+git20190320.b499514-1) ...\n",
            "Setting up libfastahack0:amd64 (1.0.0+dfsg-9) ...\n",
            "Setting up libtabixpp0:amd64 (1.1.0-6) ...\n",
            "Setting up libssw0:amd64 (1.1-13) ...\n",
            "Setting up sysstat (12.5.2-2ubuntu0.2) ...\n",
            "\n",
            "Creating config file /etc/default/sysstat with new version\n",
            "update-alternatives: using /usr/bin/sar.sysstat to provide /usr/bin/sar (sar) in auto mode\n",
            "Created symlink /etc/systemd/system/sysstat.service.wants/sysstat-collect.timer → /lib/systemd/system/sysstat-collect.timer.\n",
            "Created symlink /etc/systemd/system/sysstat.service.wants/sysstat-summary.timer → /lib/systemd/system/sysstat-summary.timer.\n",
            "Created symlink /etc/systemd/system/multi-user.target.wants/sysstat.service → /lib/systemd/system/sysstat.service.\n",
            "Setting up libseqlib2:amd64 (1.2.0+dfsg-8build1) ...\n",
            "Setting up parallel (20210822+ds-2) ...\n",
            "Setting up libvcflib1:amd64 (1.0.3+dfsg-1) ...\n",
            "Setting up freebayes (1.3.6-1) ...\n",
            "Processing triggers for man-db (2.10.2-1) ...\n",
            "Processing triggers for libc-bin (2.35-0ubuntu3.8) ...\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbb.so.12 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc.so.2 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libhwloc.so.15 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_0.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtcm.so.1 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtcm_debug.so.1 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_adapter_opencl.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_loader.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_5.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libumf.so.1 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc_proxy.so.2 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_adapter_level_zero.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_adapter_level_zero_v2.so.0 is not a symbolic link\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!apt-get install -y freebayes\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "xiWoWLA7aUgM",
        "outputId": "0b21c537-a729-40f5-c211-87e7a0f60257"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "Suggested packages:\n",
            "  libnss-mdns fonts-dejavu-extra fonts-ipafont-gothic fonts-ipafont-mincho\n",
            "  fonts-wqy-microhei | fonts-wqy-zenhei fonts-indic\n",
            "The following NEW packages will be installed:\n",
            "  openjdk-11-jre-headless\n",
            "0 upgraded, 1 newly installed, 0 to remove and 57 not upgraded.\n",
            "Need to get 42.6 MB of archives.\n",
            "After this operation, 176 MB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 openjdk-11-jre-headless amd64 11.0.29+7-1ubuntu1~22.04 [42.6 MB]\n",
            "Fetched 42.6 MB in 2s (18.2 MB/s)\n",
            "Selecting previously unselected package openjdk-11-jre-headless:amd64.\n",
            "(Reading database ... 122189 files and directories currently installed.)\n",
            "Preparing to unpack .../openjdk-11-jre-headless_11.0.29+7-1ubuntu1~22.04_amd64.deb ...\n",
            "Unpacking openjdk-11-jre-headless:amd64 (11.0.29+7-1ubuntu1~22.04) ...\n",
            "Setting up openjdk-11-jre-headless:amd64 (11.0.29+7-1ubuntu1~22.04) ...\n",
            "update-alternatives: using /usr/lib/jvm/java-11-openjdk-amd64/bin/jjs to provide /usr/bin/jjs (jjs) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-11-openjdk-amd64/bin/rmid to provide /usr/bin/rmid (rmid) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-11-openjdk-amd64/bin/pack200 to provide /usr/bin/pack200 (pack200) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-11-openjdk-amd64/bin/unpack200 to provide /usr/bin/unpack200 (unpack200) in auto mode\n"
          ]
        }
      ],
      "source": [
        "!apt-get install -y openjdk-11-jre-headless"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "xlexZX6ha0p_",
        "outputId": "ce3a08bc-6b21-4791-b698-1ffe393f06d8"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "unzip is already the newest version (6.0-26ubuntu3.2).\n",
            "0 upgraded, 0 newly installed, 0 to remove and 57 not upgraded.\n"
          ]
        }
      ],
      "source": [
        "!apt-get install -y unzip\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "SU2sh8AfbSGB",
        "outputId": "7e2e01df-d67d-4d43-d107-ad254204c2ab"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2025-12-01 00:26:33--  https://snpeff-public.s3.amazonaws.com/versions/snpEff_latest_core.zip\n",
            "Resolving snpeff-public.s3.amazonaws.com (snpeff-public.s3.amazonaws.com)... 52.92.240.41, 52.218.238.35, 52.92.130.233, ...\n",
            "Connecting to snpeff-public.s3.amazonaws.com (snpeff-public.s3.amazonaws.com)|52.92.240.41|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 66478586 (63M) [application/zip]\n",
            "Saving to: ‘snpEff_latest_core.zip’\n",
            "\n",
            "snpEff_latest_core. 100%[===================>]  63.40M  40.6MB/s    in 1.6s    \n",
            "\n",
            "2025-12-01 00:26:35 (40.6 MB/s) - ‘snpEff_latest_core.zip’ saved [66478586/66478586]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!wget https://snpeff-public.s3.amazonaws.com/versions/snpEff_latest_core.zip"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "39grZwOcbq4h",
        "outputId": "1d9aa038-aadb-43c8-b491-315c0bbb6673"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Archive:  snpEff_latest_core.zip\n",
            "   creating: snpEff/\n",
            "  inflating: snpEff/LICENSE.md       \n",
            "  inflating: snpEff/snpEff.jar       \n",
            "  inflating: snpEff/SnpSift.jar      \n",
            "   creating: snpEff/galaxy/\n",
            "  inflating: snpEff/galaxy/snpSift_int.xml  \n",
            "   creating: snpEff/galaxy/tool-data/\n",
            "  inflating: snpEff/galaxy/tool-data/snpEff_genomes.loc  \n",
            "  inflating: snpEff/galaxy/tool-data/snpEff_genomes.loc.sample  \n",
            "  inflating: snpEff/galaxy/snpEffWrapper.pl  \n",
            "  inflating: snpEff/galaxy/snpEff.xml  \n",
            "  inflating: snpEff/galaxy/tool_conf.xml  \n",
            "  inflating: snpEff/galaxy/snpSift_caseControl.xml  \n",
            "  inflating: snpEff/galaxy/snpSift_filter.xml  \n",
            "  inflating: snpEff/galaxy/snpSift_annotate.xml  \n",
            "  inflating: snpEff/galaxy/snpSiftWrapper.pl  \n",
            "  inflating: snpEff/galaxy/tool_dependencies.xml  \n",
            "  inflating: snpEff/galaxy/snpEff_download.xml  \n",
            "  inflating: snpEff/snpEff.config    \n",
            "   creating: snpEff/examples/\n",
            "  inflating: snpEff/examples/samples_cancer.txt  \n",
            "  inflating: snpEff/examples/example_motif.vcf  \n",
            "  inflating: snpEff/examples/cancer.eff.vcf  \n",
            "  inflating: snpEff/examples/examples.sh  \n",
            "  inflating: snpEff/examples/test.chr22.ann.txt  \n",
            "  inflating: snpEff/examples/test.1KG.vcf  \n",
            "  inflating: snpEff/examples/cancer.ann.vcf  \n",
            "  inflating: snpEff/examples/1kg.head_chr1.filtered.vcf.gz  \n",
            " extracting: snpEff/examples/test.vcf  \n",
            "  inflating: snpEff/examples/test.chr22.ann.filter_missense_first.vcf  \n",
            "  inflating: snpEff/examples/test.1KG.ann_reg.vcf  \n",
            "  inflating: snpEff/examples/cancer.vcf  \n",
            "  inflating: snpEff/examples/my_annotations.bed  \n",
            "  inflating: snpEff/examples/test.chr22.ann.filter_missense_any_TRMT2A.vcf  \n",
            "  inflating: snpEff/examples/cancer_pedigree.ann.vcf  \n",
            "  inflating: snpEff/examples/variants_2.vcf  \n",
            "  inflating: snpEff/examples/test.ann.vcf  \n",
            "  inflating: snpEff/examples/intervals.bed  \n",
            "  inflating: snpEff/examples/test.1KG.ann_encode.vcf  \n",
            "  inflating: snpEff/examples/variants_1.vcf  \n",
            "  inflating: snpEff/examples/test.chr22.ann.vcf  \n",
            "  inflating: snpEff/examples/file.vcf  \n",
            "  inflating: snpEff/examples/variants_1.ann.vcf  \n",
            "  inflating: snpEff/examples/test.chr22.vcf  \n",
            "  inflating: snpEff/examples/1kg.head_chr1.vcf.gz  \n",
            "  inflating: snpEff/examples/samples_cancer_one.txt  \n",
            "  inflating: snpEff/examples/test.chr22.ann.filter_missense_any.vcf  \n",
            "  inflating: snpEff/examples/test.chr22.ann.one_per_line.txt  \n",
            "  inflating: snpEff/examples/variants_2.ann.vcf  \n",
            "  inflating: snpEff/examples/cancer_pedigree.vcf  \n",
            "  inflating: snpEff/examples/test.chr22.ann.filter_missense.vcf  \n",
            "   creating: snpEff/scripts/\n",
            "   creating: snpEff/scripts/gsa/\n",
            "  inflating: snpEff/scripts/gsa/bayesFactor_correction_scoreCount_max10.sh  \n",
            "  inflating: snpEff/scripts/gsa/geneSetOverlap.sort.txt  \n",
            "  inflating: snpEff/scripts/gsa/create_sets.bds  \n",
            "  inflating: snpEff/scripts/gsa/bayesFactor_correction_scoreCount.r  \n",
            "  inflating: snpEff/scripts/gsa/geneSetsGtex.py  \n",
            "  inflating: snpEff/scripts/gsa/pvalue_correction_scoreCount.sh  \n",
            "  inflating: snpEff/scripts/gsa/pvalue_correction_scoreCount.r  \n",
            "  inflating: snpEff/scripts/gsa/checkGeneNames.py  \n",
            "  inflating: snpEff/scripts/gsa/geneSetOverlap.py  \n",
            "  inflating: snpEff/scripts/gsa/pvalue_correction_scoreCount_min10.sh  \n",
            "  inflating: snpEff/scripts/gsa/bayesFactor_correction_scoreCount.sh  \n",
            "  inflating: snpEff/scripts/make_dbNSFP.sh  \n",
            "  inflating: snpEff/scripts/snpEff   \n",
            "  inflating: snpEff/scripts/snpSift  \n",
            "   creating: snpEff/scripts/_OLD/\n",
            "  inflating: snpEff/scripts/_OLD/vcfBareBones.pl  \n",
            "  inflating: snpEff/scripts/_OLD/cgShore.pl  \n",
            "  inflating: snpEff/scripts/_OLD/vcfInfoOnePerLine.pl  \n",
            "  inflating: snpEff/scripts/_OLD/sam2fastq.pl  \n",
            "  inflating: snpEff/scripts/_OLD/queue.pl  \n",
            "  inflating: snpEff/scripts/_OLD/extractSequences.pl  \n",
            "  inflating: snpEff/scripts/_OLD/isutf8.py  \n",
            "  inflating: snpEff/scripts/_OLD/db.pl  \n",
            "  inflating: snpEff/scripts/_OLD/annotate_demo.sh  \n",
            "  inflating: snpEff/scripts/_OLD/filterBy.py  \n",
            "  inflating: snpEff/scripts/_OLD/countColumns.py  \n",
            "  inflating: snpEff/scripts/_OLD/plotSmoothScatter.pl  \n",
            "  inflating: snpEff/scripts/_OLD/plot.pl  \n",
            "  inflating: snpEff/scripts/_OLD/ped2vcf.py  \n",
            "  inflating: snpEff/scripts/_OLD/uniqCut.pl  \n",
            "  inflating: snpEff/scripts/_OLD/vcfEffOnePerLine.pl  \n",
            "  inflating: snpEff/scripts/_OLD/joinSnpEff.pl  \n",
            "  inflating: snpEff/scripts/_OLD/vcfEffHighest.ORI.py  \n",
            "  inflating: snpEff/scripts/_OLD/fastqSplit.pl  \n",
            "  inflating: snpEff/scripts/_OLD/splitChr.pl  \n",
            "  inflating: snpEff/scripts/_OLD/make_and_commit_docs.sh  \n",
            "  inflating: snpEff/scripts/_OLD/join.pl  \n",
            "  inflating: snpEff/scripts/_OLD/txt2fa.pl  \n",
            "  inflating: snpEff/scripts/_OLD/nextProt_filter.pl  \n",
            "  inflating: snpEff/scripts/_OLD/vcfOnlyAlts.pl  \n",
            "  inflating: snpEff/scripts/_OLD/annotate_demo_GATK.sh  \n",
            "  inflating: snpEff/scripts/_OLD/fastaSplit.pl  \n",
            "  inflating: snpEff/scripts/_OLD/transpose.pl  \n",
            "  inflating: snpEff/scripts/_OLD/sortLine.py  \n",
            "  inflating: snpEff/scripts/_OLD/vcfAnnFirst.py  \n",
            "  inflating: snpEff/scripts/_OLD/cgShore.sh  \n",
            "  inflating: snpEff/scripts/_OLD/fasta2tab.pl  \n",
            "  inflating: snpEff/scripts/_OLD/txt2vcf.py  \n",
            "  inflating: snpEff/scripts/_OLD/snpSift_filter_sample_to_number.pl  \n",
            "  inflating: snpEff/scripts/_OLD/uniqCount.pl  \n",
            "  inflating: snpEff/scripts/_OLD/swapCols.pl  \n",
            "  inflating: snpEff/scripts/_OLD/fastaSample.pl  \n",
            "  inflating: snpEff/scripts/_OLD/plotQQ.pl  \n",
            "  inflating: snpEff/scripts/_OLD/bedEffOnePerLine.pl  \n",
            "  inflating: snpEff/scripts/_OLD/vcfRefCorrect.py  \n",
            "  inflating: snpEff/scripts/_OLD/plotQQsubsample.pl  \n",
            "  inflating: snpEff/scripts/_OLD/sample.pl  \n",
            "  inflating: snpEff/scripts/_OLD/vcfFilterSamples.pl  \n",
            "  inflating: snpEff/scripts/_OLD/1kg.sh  \n",
            "  inflating: snpEff/scripts/_OLD/wigSplit.pl  \n",
            "  inflating: snpEff/scripts/_OLD/gffRemovePhase.pl  \n",
            "  inflating: snpEff/scripts/_OLD/plotLabel.pl  \n",
            "  inflating: snpEff/scripts/_OLD/vcfReduceGenotypes.pl  \n",
            "  inflating: snpEff/scripts/_OLD/plotMA.pl  \n",
            "  inflating: snpEff/scripts/_OLD/statsNum.pl  \n",
            "  inflating: snpEff/scripts/_OLD/plotHistogram.pl  \n",
            "  inflating: snpEff/scripts/_OLD/plotXY.pl  \n",
            "  inflating: snpEff/scripts/buildDbNcbi.sh  \n",
            "   creating: snpEff/exec/\n",
            "  inflating: snpEff/exec/snpeff      \n",
            "  inflating: snpEff/exec/snpsift     \n"
          ]
        }
      ],
      "source": [
        "!unzip -o snpEff_latest_core.zip\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "P3lMr4GIc_ie",
        "outputId": "4ef11163-935e-4152-9a09-422b64c2d033"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Cloning into 'bioinfotest'...\n",
            "remote: Enumerating objects: 25, done.\u001b[K\n",
            "remote: Counting objects: 100% (25/25), done.\u001b[K\n",
            "remote: Compressing objects: 100% (21/21), done.\u001b[K\n",
            "remote: Total 25 (delta 4), reused 8 (delta 0), pack-reused 0 (from 0)\u001b[K\n",
            "Receiving objects: 100% (25/25), 12.15 MiB | 40.79 MiB/s, done.\n",
            "Resolving deltas: 100% (4/4), done.\n"
          ]
        }
      ],
      "source": [
        "!git clone https://github.com/danyliramenezes/bioinfotest"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "FUylN4mhdwgw",
        "outputId": "02d57174-03f8-4018-ed9a-304457d1ceee"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2025-12-01 00:26:46--  http://hgdownload.soe.ucsc.edu/goldenPath/hg19/bigZips/hg19.fa.gz\n",
            "Resolving hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)... 128.114.119.163\n",
            "Connecting to hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)|128.114.119.163|:80... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 948731419 (905M) [application/x-gzip]\n",
            "Saving to: ‘bioinfotest/data/hg19.fasta.gz’\n",
            "\n",
            "bioinfotest/data/hg 100%[===================>] 904.78M  57.5MB/s    in 15s     \n",
            "\n",
            "2025-12-01 00:27:02 (59.8 MB/s) - ‘bioinfotest/data/hg19.fasta.gz’ saved [948731419/948731419]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!wget http://hgdownload.soe.ucsc.edu/goldenPath/hg19/bigZips/hg19.fa.gz -O bioinfotest/data/hg19.fasta.gz\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "XEiQfAm1d3-Z"
      },
      "outputs": [],
      "source": [
        "!gunzip bioinfotest/data/hg19.fasta.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "WeTRTdZDeb7f",
        "outputId": "a5122bd6-bd43-4b92-b5b0-3160c584f9b2"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[bwa_index] Pack FASTA... 15.98 sec\n",
            "[bwa_index] Construct BWT for the packed sequence...\n",
            "[BWTIncCreate] textLength=6274322528, availableWord=453484340\n",
            "[BWTIncConstructFromPacked] 10 iterations done. 100000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 20 iterations done. 200000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 30 iterations done. 300000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 40 iterations done. 400000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 50 iterations done. 500000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 60 iterations done. 600000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 70 iterations done. 700000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 80 iterations done. 800000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 90 iterations done. 900000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 100 iterations done. 1000000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 110 iterations done. 1100000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 120 iterations done. 1200000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 130 iterations done. 1300000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 140 iterations done. 1400000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 150 iterations done. 1500000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 160 iterations done. 1600000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 170 iterations done. 1700000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 180 iterations done. 1800000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 190 iterations done. 1900000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 200 iterations done. 2000000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 210 iterations done. 2100000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 220 iterations done. 2200000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 230 iterations done. 2300000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 240 iterations done. 2400000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 250 iterations done. 2500000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 260 iterations done. 2600000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 270 iterations done. 2700000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 280 iterations done. 2800000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 290 iterations done. 2900000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 300 iterations done. 3000000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 310 iterations done. 3100000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 320 iterations done. 3200000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 330 iterations done. 3300000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 340 iterations done. 3400000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 350 iterations done. 3500000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 360 iterations done. 3600000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 370 iterations done. 3700000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 380 iterations done. 3800000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 390 iterations done. 3900000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 400 iterations done. 4000000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 410 iterations done. 4100000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 420 iterations done. 4200000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 430 iterations done. 4300000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 440 iterations done. 4400000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 450 iterations done. 4500000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 460 iterations done. 4600000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 470 iterations done. 4700000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 480 iterations done. 4800000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 490 iterations done. 4900000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 500 iterations done. 5000000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 510 iterations done. 5100000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 520 iterations done. 5200000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 530 iterations done. 5300000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 540 iterations done. 5400000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 550 iterations done. 5500000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 560 iterations done. 5600000000 characters processed.\n",
            "[BWTIncConstructFromPacked] 570 iterations done. 5694239600 characters processed.\n",
            "[BWTIncConstructFromPacked] 580 iterations done. 5777996048 characters processed.\n",
            "[BWTIncConstructFromPacked] 590 iterations done. 5852434944 characters processed.\n",
            "[BWTIncConstructFromPacked] 600 iterations done. 5918592432 characters processed.\n",
            "[BWTIncConstructFromPacked] 610 iterations done. 5977389360 characters processed.\n",
            "[BWTIncConstructFromPacked] 620 iterations done. 6029644208 characters processed.\n",
            "[BWTIncConstructFromPacked] 630 iterations done. 6076084432 characters processed.\n",
            "[BWTIncConstructFromPacked] 640 iterations done. 6117356576 characters processed.\n",
            "[BWTIncConstructFromPacked] 650 iterations done. 6154035344 characters processed.\n",
            "[BWTIncConstructFromPacked] 660 iterations done. 6186631520 characters processed.\n",
            "[BWTIncConstructFromPacked] 670 iterations done. 6215599040 characters processed.\n",
            "[BWTIncConstructFromPacked] 680 iterations done. 6241341392 characters processed.\n",
            "[BWTIncConstructFromPacked] 690 iterations done. 6264217232 characters processed.\n",
            "[bwt_gen] Finished constructing BWT in 695 iterations.\n",
            "[bwa_index] 3831.01 seconds elapse.\n",
            "[bwa_index] Update BWT... 14.42 sec\n",
            "[bwa_index] Pack forward-only FASTA... 12.76 sec\n",
            "[bwa_index] Construct SA from BWT and Occ... 3138.31 sec\n",
            "[main] Version: 0.7.17-r1188\n",
            "[main] CMD: bwa index bioinfotest/data/hg19.fasta\n",
            "[main] Real time: 7242.441 sec; CPU: 7012.487 sec\n"
          ]
        }
      ],
      "source": [
        "!bwa index bioinfotest/data/hg19.fasta"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!apt-get install -y samtools\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ru9s6HIpSP0_",
        "outputId": "c461c412-9526-4458-f73f-797104bbd127"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "The following additional packages will be installed:\n",
            "  libhts3 libhtscodecs2\n",
            "Suggested packages:\n",
            "  cwltool\n",
            "The following NEW packages will be installed:\n",
            "  libhts3 libhtscodecs2 samtools\n",
            "0 upgraded, 3 newly installed, 0 to remove and 41 not upgraded.\n",
            "Need to get 963 kB of archives.\n",
            "After this operation, 2,270 kB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libhtscodecs2 amd64 1.1.1-3 [53.2 kB]\n",
            "Get:2 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libhts3 amd64 1.13+ds-2build1 [390 kB]\n",
            "Get:3 http://archive.ubuntu.com/ubuntu jammy/universe amd64 samtools amd64 1.13-4 [520 kB]\n",
            "Fetched 963 kB in 1s (907 kB/s)\n",
            "Selecting previously unselected package libhtscodecs2:amd64.\n",
            "(Reading database ... 121713 files and directories currently installed.)\n",
            "Preparing to unpack .../libhtscodecs2_1.1.1-3_amd64.deb ...\n",
            "Unpacking libhtscodecs2:amd64 (1.1.1-3) ...\n",
            "Selecting previously unselected package libhts3:amd64.\n",
            "Preparing to unpack .../libhts3_1.13+ds-2build1_amd64.deb ...\n",
            "Unpacking libhts3:amd64 (1.13+ds-2build1) ...\n",
            "Selecting previously unselected package samtools.\n",
            "Preparing to unpack .../samtools_1.13-4_amd64.deb ...\n",
            "Unpacking samtools (1.13-4) ...\n",
            "Setting up libhtscodecs2:amd64 (1.1.1-3) ...\n",
            "Setting up libhts3:amd64 (1.13+ds-2build1) ...\n",
            "Setting up samtools (1.13-4) ...\n",
            "Processing triggers for man-db (2.10.2-1) ...\n",
            "Processing triggers for libc-bin (2.35-0ubuntu3.8) ...\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbb.so.12 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc.so.2 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libhwloc.so.15 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_0.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtcm.so.1 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtcm_debug.so.1 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_adapter_opencl.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_loader.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_5.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libumf.so.1 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc_proxy.so.2 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_adapter_level_zero.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_adapter_level_zero_v2.so.0 is not a symbolic link\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!ls -lh bioinfotest/data"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "0cJQcY83SgsX",
        "outputId": "71554b3b-c700-4f9b-b7c9-98b59e518031"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "total 13M\n",
            "-rw-r--r-- 1 root root 6.0M Dec  1 00:24 510-7-BRCA_S8_L001_R1_001.fastq.gz\n",
            "-rw-r--r-- 1 root root 6.2M Dec  1 00:24 510-7-BRCA_S8_L001_R2_001.fastq.gz\n",
            "-rw-r--r-- 1 root root  213 Dec  1 00:24 desconhecido.fasta\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!samtools faidx bioinfotest/data/hg19.fasta"
      ],
      "metadata": {
        "id": "ERWn0E99vyeo"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!ls -lh bioinfotest/data"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "OwN9Hrj0v6ST",
        "outputId": "075ede82-cf9b-4374-dc4e-5125002365e7"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "total 8.2G\n",
            "-rw-r--r-- 1 root root 6.0M Dec  1 00:24 510-7-BRCA_S8_L001_R1_001.fastq.gz\n",
            "-rw-r--r-- 1 root root 6.2M Dec  1 00:24 510-7-BRCA_S8_L001_R2_001.fastq.gz\n",
            "-rw-r--r-- 1 root root  213 Dec  1 00:24 desconhecido.fasta\n",
            "-rw-r--r-- 1 root root 3.0G Aug 21  2018 hg19.fasta\n",
            "-rw-r--r-- 1 root root 8.4K Dec  1 01:35 hg19.fasta.amb\n",
            "-rw-r--r-- 1 root root 4.0K Dec  1 01:35 hg19.fasta.ann\n",
            "-rw-r--r-- 1 root root 3.0G Dec  1 01:34 hg19.fasta.bwt\n",
            "-rw-r--r-- 1 root root 3.5K Dec  1 02:30 hg19.fasta.fai\n",
            "-rw-r--r-- 1 root root 748M Dec  1 01:35 hg19.fasta.pac\n",
            "-rw-r--r-- 1 root root 1.5G Dec  1 02:28 hg19.fasta.sa\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "SAMPLE=\"510-7-BRCA\"\n",
        "REF=\"bioinfotest/data/hg19.fasta\"\n",
        "R1=\"bioinfotest/data/510-7-BRCA_S8_L001_R1_001.fastq.gz\"\n",
        "R2=\"bioinfotest/data/510-7-BRCA_S8_L001_R2_001.fastq.gz\"\n",
        "\n",
        "!bwa mem -t 4 $REF $R1 $R2 > bioinfotest/output/${SAMPLE}.sam\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "2uo9CEpTxuUP",
        "outputId": "7603a6d0-9bfc-45f2-cc30-437bccfba2a2"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[M::process] read 128552 sequences (17254588 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (9, 61182, 0, 8)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (123, 200, 324)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 726)\n",
            "[M::mem_pestat] mean and std.dev: (233.64, 136.65)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (1, 927)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 128552 reads in 44.389 CPU sec, 25.046 real sec\n",
            "[main] Version: 0.7.17-r1188\n",
            "[main] CMD: bwa mem -t 4 bioinfotest/data/hg19.fasta bioinfotest/data/510-7-BRCA_S8_L001_R1_001.fastq.gz bioinfotest/data/510-7-BRCA_S8_L001_R2_001.fastq.gz\n",
            "[main] Real time: 59.440 sec; CPU: 54.738 sec\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "SAMPLE=\"510-7-BRCA\"\n",
        "\n",
        "!samtools view -bS bioinfotest/output/${SAMPLE}.sam | samtools sort -o bioinfotest/output/${SAMPLE}.sorted.bam\n",
        "!samtools index bioinfotest/output/${SAMPLE}.sorted.bam\n",
        "\n",
        "!ls -lh bioinfotest/output\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "sum0efCfyJuZ",
        "outputId": "dda42418-f9fc-45f6-a5e2-9b181be3f0d9"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "total 58M\n",
            "-rw-r--r-- 1 root root  50M Dec  1 02:39 10-7-BRCA.sam\n",
            "-rw-r--r-- 1 root root 7.3M Dec  1 02:40 10-7-BRCA.sorted.bam\n",
            "-rw-r--r-- 1 root root 1.5M Dec  1 02:40 10-7-BRCA.sorted.bam.bai\n",
            "-rw-r--r-- 1 root root 1.1K Dec  1 00:24 questões_2.md\n",
            "-rw-r--r-- 1 root root  892 Dec  1 00:24 questões.md\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "%%bash\n",
        "cat <<EOF > bioinfotest/data/targets.bed\n",
        "chr17   43044000    43046000    BRCA1_test_region\n",
        "EOF\n",
        "\n"
      ],
      "metadata": {
        "id": "RuAkC6iNyhMz"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!cat bioinfotest/data/targets.bed"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "zwsdczlVzXAm",
        "outputId": "05ba8829-9986-4026-8af3-4edf55ac433e"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "chr17   43044000    43046000    BRCA1_test_region\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "REF=\"bioinfotest/data/hg19.fasta\"\n",
        "BAM=\"bioinfotest/output/10-7-BRCA.sorted.bam\"\n",
        "BED=\"bioinfotest/data/targets.bed\"\n",
        "VCF_OUT=\"bioinfotest/output/10-7-BRCA.raw.vcf\"\n",
        "\n",
        "!freebayes -f $REF --targets $BED $BAM > $VCF_OUT\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "cIjFfyRiyken",
        "outputId": "bf7c5e30-158c-4144-d1d2-3bd4c010f2e7"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "ERROR(freebayes): Invalid target region coordinates (chr17 0 0) right bound is lower than left bound!\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# Criar arquivo BED com BRCA1 e BRCA2\n",
        "with open('bioinfotest/data/BRCA.bed', 'w') as f:\n",
        "    f.write(\"chr17\\t41196312\\t41277500\\n\")  # BRCA1\n",
        "    f.write(\"chr13\\t32889617\\t32973809\\n\")  # BRCA2\n",
        "\n",
        "!echo \"=== Verificando arquivo BED ===\"\n",
        "!cat bioinfotest/data/BRCA.bed"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "riMUjHZUzqRV",
        "outputId": "706f9ad2-8ed4-411c-e697-87d29e5564b2"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "=== Verificando arquivo BED ===\n",
            "chr17\t41196312\t41277500\n",
            "chr13\t32889617\t32973809\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!cat -n bioinfotest/data/alvos.bed"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "IFbn3kJM2NC2",
        "outputId": "26ca4d86-e004-474f-9df8-5794f48d2c30"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "     1\tchr1\t35000\t36000\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!samtools idxstats bioinfotest/output/10-7-BRCA.sorted.bam | head\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "0kqISTeI01Hn",
        "outputId": "74fee908-6da0-4ddc-f3c3-ea9f931844b2"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "chr1\t249250621\t954\t2\n",
            "chr2\t243199373\t265\t3\n",
            "chr3\t198022430\t704\t1\n",
            "chr4\t191154276\t279\t1\n",
            "chr5\t180915260\t468\t2\n",
            "chr6\t171115067\t767\t0\n",
            "chr7\t159138663\t313\t0\n",
            "chrX\t155270560\t184\t0\n",
            "chr8\t146364022\t157\t0\n",
            "chr9\t141213431\t596\t2\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "REF=\"bioinfotest/data/hg19.fasta\"\n",
        "BAM=\"bioinfotest/output/10-7-BRCA.sorted.bam\"\n",
        "BED=\"bioinfotest/data/alvos.bed\"\n",
        "VCF_OUT=\"bioinfotest/output/10-7-BRCA.raw.vcf\"\n",
        "\n",
        "!freebayes -f $REF --targets $BED $BAM > $VCF_OUT"
      ],
      "metadata": {
        "id": "2z2cs1dS3ICH"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!apt-get update -y\n",
        "!apt-get install -y openjdk-21-jre-headless"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "eeBy0ChY4qYw",
        "outputId": "31e4b62b-1156-40c3-9355-4e3a7fd6a85e"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\r0% [Working]\r            \rHit:1 http://archive.ubuntu.com/ubuntu jammy InRelease\n",
            "\r0% [Connecting to security.ubuntu.com (185.125.190.81)] [Connecting to cloud.r-\r                                                                               \rHit:2 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  InRelease\n",
            "\r                                                                               \rHit:3 https://cli.github.com/packages stable InRelease\n",
            "\r0% [Waiting for headers] [Connecting to security.ubuntu.com (185.125.190.81)] [\r                                                                               \rHit:4 http://archive.ubuntu.com/ubuntu jammy-updates InRelease\n",
            "\r0% [Waiting for headers] [Connecting to security.ubuntu.com (185.125.190.81)] [\r                                                                               \rHit:5 http://archive.ubuntu.com/ubuntu jammy-backports InRelease\n",
            "\r0% [Waiting for headers] [Connected to cloud.r-project.org (13.35.78.126)] [Wai\r                                                                               \rHit:6 https://r2u.stat.illinois.edu/ubuntu jammy InRelease\n",
            "\r0% [Waiting for headers] [Waiting for headers] [Connecting to ppa.launchpadcont\r                                                                               \rHit:7 https://cloud.r-project.org/bin/linux/ubuntu jammy-cran40/ InRelease\n",
            "Hit:8 http://security.ubuntu.com/ubuntu jammy-security InRelease\n",
            "Hit:9 https://ppa.launchpadcontent.net/deadsnakes/ppa/ubuntu jammy InRelease\n",
            "Hit:10 https://ppa.launchpadcontent.net/graphics-drivers/ppa/ubuntu jammy InRelease\n",
            "Hit:11 https://ppa.launchpadcontent.net/ubuntugis/ppa/ubuntu jammy InRelease\n",
            "Reading package lists... Done\n",
            "W: Skipping acquire of configured file 'main/source/Sources' as repository 'https://r2u.stat.illinois.edu/ubuntu jammy InRelease' does not seem to provide it (sources.list entry misspelt?)\n",
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "Suggested packages:\n",
            "  libnss-mdns fonts-dejavu-extra fonts-ipafont-gothic fonts-ipafont-mincho\n",
            "  fonts-wqy-microhei | fonts-wqy-zenhei fonts-indic\n",
            "The following NEW packages will be installed:\n",
            "  openjdk-21-jre-headless\n",
            "0 upgraded, 1 newly installed, 0 to remove and 57 not upgraded.\n",
            "Need to get 46.9 MB of archives.\n",
            "After this operation, 203 MB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 openjdk-21-jre-headless amd64 21.0.9+10-1~22.04 [46.9 MB]\n",
            "Fetched 46.9 MB in 3s (18.6 MB/s)\n",
            "Selecting previously unselected package openjdk-21-jre-headless:amd64.\n",
            "(Reading database ... 122524 files and directories currently installed.)\n",
            "Preparing to unpack .../openjdk-21-jre-headless_21.0.9+10-1~22.04_amd64.deb ...\n",
            "Unpacking openjdk-21-jre-headless:amd64 (21.0.9+10-1~22.04) ...\n",
            "Setting up openjdk-21-jre-headless:amd64 (21.0.9+10-1~22.04) ...\n",
            "update-alternatives: using /usr/lib/jvm/java-21-openjdk-amd64/bin/java to provide /usr/bin/java (java) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-21-openjdk-amd64/bin/jpackage to provide /usr/bin/jpackage (jpackage) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-21-openjdk-amd64/bin/keytool to provide /usr/bin/keytool (keytool) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-21-openjdk-amd64/bin/rmiregistry to provide /usr/bin/rmiregistry (rmiregistry) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-21-openjdk-amd64/lib/jexec to provide /usr/bin/jexec (jexec) in auto mode\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "VCF_IN=\"bioinfotest/output/10-7-BRCA.raw.vcf\"\n",
        "VCF_ANN=\"bioinfotest/output/10-7-BRCA.annotated.vcf\"\n",
        "\n",
        "!java -Xmx4g -jar snpEff/snpEff.jar hg19 $VCF_IN > $VCF_ANN\n"
      ],
      "metadata": {
        "id": "6h1EVp364Iqs"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "# Contar sequências\n",
        "!./seqkit stats bioinfotest/data/510-7-BRCA_S8_L001_R1_001.fastq.gz\n",
        "!./seqkit stats bioinfotest/data/510-7-BRCA_S8_L001_R2_001.fastq.gz"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "kZXviDG_52yX",
        "outputId": "f1e2856a-3e31-4fc4-88cc-0bac57755ebe"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2025-12-01 03:26:38--  https://github.com/shenwei356/seqkit/releases/download/v2.3.1/seqkit_linux_amd64.tar.gz\n",
            "Resolving github.com (github.com)... 140.82.114.3\n",
            "Connecting to github.com (github.com)|140.82.114.3|:443... connected.\n",
            "HTTP request sent, awaiting response... 302 Found\n",
            "Location: https://release-assets.githubusercontent.com/github-production-release-asset/52715040/9190031d-da9a-4da6-a70f-27bea620cb98?sp=r&sv=2018-11-09&sr=b&spr=https&se=2025-12-01T04%3A07%3A57Z&rscd=attachment%3B+filename%3Dseqkit_linux_amd64.tar.gz&rsct=application%2Foctet-stream&skoid=96c2d410-5711-43a1-aedd-ab1947aa7ab0&sktid=398a6654-997b-47e9-b12b-9515b896b4de&skt=2025-12-01T03%3A07%3A24Z&ske=2025-12-01T04%3A07%3A57Z&sks=b&skv=2018-11-09&sig=FDAxq692vuoTyGdtE1eoAwWkkBA0UvoH8GpOjCxsBig%3D&jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmVsZWFzZS1hc3NldHMuZ2l0aHVidXNlcmNvbnRlbnQuY29tIiwia2V5Ijoia2V5MSIsImV4cCI6MTc2NDU1OTg5OCwibmJmIjoxNzY0NTU5NTk4LCJwYXRoIjoicmVsZWFzZWFzc2V0cHJvZHVjdGlvbi5ibG9iLmNvcmUud2luZG93cy5uZXQifQ.0SOsuWXH_wz-vdaKJYVM8veUTwtZqorX2PSctQgegZs&response-content-disposition=attachment%3B%20filename%3Dseqkit_linux_amd64.tar.gz&response-content-type=application%2Foctet-stream [following]\n",
            "--2025-12-01 03:26:38--  https://release-assets.githubusercontent.com/github-production-release-asset/52715040/9190031d-da9a-4da6-a70f-27bea620cb98?sp=r&sv=2018-11-09&sr=b&spr=https&se=2025-12-01T04%3A07%3A57Z&rscd=attachment%3B+filename%3Dseqkit_linux_amd64.tar.gz&rsct=application%2Foctet-stream&skoid=96c2d410-5711-43a1-aedd-ab1947aa7ab0&sktid=398a6654-997b-47e9-b12b-9515b896b4de&skt=2025-12-01T03%3A07%3A24Z&ske=2025-12-01T04%3A07%3A57Z&sks=b&skv=2018-11-09&sig=FDAxq692vuoTyGdtE1eoAwWkkBA0UvoH8GpOjCxsBig%3D&jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmVsZWFzZS1hc3NldHMuZ2l0aHVidXNlcmNvbnRlbnQuY29tIiwia2V5Ijoia2V5MSIsImV4cCI6MTc2NDU1OTg5OCwibmJmIjoxNzY0NTU5NTk4LCJwYXRoIjoicmVsZWFzZWFzc2V0cHJvZHVjdGlvbi5ibG9iLmNvcmUud2luZG93cy5uZXQifQ.0SOsuWXH_wz-vdaKJYVM8veUTwtZqorX2PSctQgegZs&response-content-disposition=attachment%3B%20filename%3Dseqkit_linux_amd64.tar.gz&response-content-type=application%2Foctet-stream\n",
            "Resolving release-assets.githubusercontent.com (release-assets.githubusercontent.com)... 185.199.111.133, 185.199.109.133, 185.199.110.133, ...\n",
            "Connecting to release-assets.githubusercontent.com (release-assets.githubusercontent.com)|185.199.111.133|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 7285731 (6.9M) [application/octet-stream]\n",
            "Saving to: ‘seqkit_linux_amd64.tar.gz.3’\n",
            "\n",
            "seqkit_linux_amd64. 100%[===================>]   6.95M  --.-KB/s    in 0.08s   \n",
            "\n",
            "2025-12-01 03:26:38 (88.0 MB/s) - ‘seqkit_linux_amd64.tar.gz.3’ saved [7285731/7285731]\n",
            "\n",
            "file                                                 format  type  num_seqs    sum_len  min_len  avg_len  max_len\n",
            "bioinfotest/data/510-7-BRCA_S8_L001_R1_001.fastq.gz  FASTQ   DNA     64,276  8,626,127       32    134.2      151\n",
            "file                                                 format  type  num_seqs    sum_len  min_len  avg_len  max_len\n",
            "bioinfotest/data/510-7-BRCA_S8_L001_R2_001.fastq.gz  FASTQ   DNA     64,276  8,628,461       32    134.2      151\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# Questão 2: Contar contigs no genoma hg19\n",
        "REF = \"bioinfotest/data/hg19.fasta\"\n",
        "!echo \"Número de contigs em hg19:\"\n",
        "!grep -c \"^>\" $REF"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "P0GfU0sE63u1",
        "outputId": "ba717fd0-e344-4ea1-b8ee-93badf7a1da6"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Número de contigs em hg19:\n",
            "93\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# Questão 3: Contar alinhamentos na região chr17:41197694-41197819\n",
        "BAM = \"bioinfotest/output/10-7-BRCA.sorted.bam\"\n",
        "!echo \"Alinhamentos na região chr17:41197694-41197819:\"\n",
        "!samtools view -c $BAM chr17:41197694-41197819"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Oia1--uG7NLh",
        "outputId": "c3f495e3-d8f0-4e6b-dbec-6ecb29c71f9f"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Alinhamentos na região chr17:41197694-41197819:\n",
            "2211\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "BAM = \"bioinfotest/output/10-7-BRCA.sorted.bam\"\n",
        "!echo \"Alinhamentos não mapeados:\"\n",
        "!samtools view -c -f 4 $BAM"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "hZ0D8KLa7UUq",
        "outputId": "739f2006-1b32-481e-9180-3e4b5a89056f"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Alinhamentos não mapeados:\n",
            "2663\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "VCF_IN=\"bioinfotest/output/10-7-BRCA.raw.vcf\"\n",
        "VCF_ANN=\"bioinfotest/output/10-7-BRCA.annotated.vcf\"\n",
        "\n",
        "!java -Xmx4g -jar snpEff/snpEff.jar hg19 $VCF_IN > $VCF_ANN"
      ],
      "metadata": {
        "id": "uNwUvpRo9srf"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "# Contar variantes por impacto\n",
        "VCF_ANN = \"bioinfotest/output/10-7-BRCA.annotated.vcf\"\n",
        "\n",
        "print(\"=== CONTAGEM DE VARIANTES POR IMPACTO ===\\n\")\n",
        "\n",
        "# HIGH\n",
        "!echo \"Variantes HIGH:\"\n",
        "!grep -v \"^#\" $VCF_ANN | grep -o \"HIGH\" | wc -l\n",
        "\n",
        "# MODERATE\n",
        "!echo \"Variantes MODERATE:\"\n",
        "!grep -v \"^#\" $VCF_ANN | grep -o \"MODERATE\" | wc -l\n",
        "\n",
        "# LOW\n",
        "!echo \"Variantes LOW:\"\n",
        "!grep -v \"^#\" $VCF_ANN | grep -o \"LOW\" | wc -l\n",
        "\n",
        "print(\"\\n=== DETALHES DAS VARIANTES HIGH ===\\n\")\n",
        "!echo \"=== Variantes HIGH ===\"\n",
        "!grep \"HIGH\" $VCF_ANN | grep -v \"^#\" | cut -f1,2,4,5\n",
        "# Mostrar detalhes das variantes HIGH\n",
        "!echo \"\"\n",
        "!echo \"=== Detalhes das variantes HIGH ===\"\n",
        "!grep -v \"^#\" $VCF_ANN | grep \"HIGH\" | awk '{print \"Cromossomo: \"$1\"\\nPosição: \"$2\"\\nRef: \"$4\"\\nAlt: \"$5\"\\n---\"}'\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "pLR9cyQz7jLr",
        "outputId": "f5b1d42e-99cc-4d64-fa44-3e5ce1119a88"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "=== CONTAGEM DE VARIANTES POR IMPACTO ===\n",
            "\n",
            "Variantes HIGH:\n",
            "5\n",
            "Variantes MODERATE:\n",
            "2\n",
            "Variantes LOW:\n",
            "8\n",
            "\n",
            "=== DETALHES DAS VARIANTES HIGH ===\n",
            "\n",
            "=== Variantes HIGH ===\n",
            "chr17\t41222948\tTTCTTCTGGGGTCAGGCCAGA\tTA\n",
            "\n",
            "=== Detalhes das variantes HIGH ===\n",
            "^C\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# Criar arquivo BED correto\n",
        "with open('bioinfotest/data/BRCA.bed', 'w') as f:\n",
        "    f.write(\"chr17\\t41196312\\t41277500\\tBRCA1\\n\")\n",
        "    f.write(\"chr13\\t32889617\\t32973809\\tBRCA2\\n\")\n",
        "\n",
        "print(\"Arquivo BRCA.bed criado com sucesso!\")\n",
        "\n",
        "# Verificar o conteúdo\n",
        "!echo \"=== Conteúdo do arquivo BRCA.bed ===\"\n",
        "!cat bioinfotest/data/BRCA.bed\n",
        "\n",
        "# Verificar formato (deve ter TABs)\n",
        "!echo -e \"\\n=== Verificando formato (TABs) ===\"\n",
        "!cat -A bioinfotest/data/BRCA.bed"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Ruvwwp5t9RBS",
        "outputId": "5e5db7bf-fd6a-4265-dda2-0ccaf8ef7f53"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Arquivo BRCA.bed criado com sucesso!\n",
            "=== Conteúdo do arquivo BRCA.bed ===\n",
            "chr17\t41196312\t41277500\tBRCA1\n",
            "chr13\t32889617\t32973809\tBRCA2\n",
            "\n",
            "=== Verificando formato (TABs) ===\n",
            "chr17^I41196312^I41277500^IBRCA1$\n",
            "chr13^I32889617^I32973809^IBRCA2$\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# Ver o conteúdo do arquivo de alvos\n",
        "BED = \"bioinfotest/data/BRCA.bed\"\n",
        "!echo \"=== Conteúdo do arquivo BRCA.list ===\"\n",
        "!cat $BED\n",
        "\n",
        "!echo -e \"\\n=== Verificar formato (deve ter TABs) ===\"\n",
        "!cat -A $BED"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "1rQzH84D8MqP",
        "outputId": "64d11ced-170e-4463-d4a5-ee2b488c00b0"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "=== Conteúdo do arquivo BRCA.list ===\n",
            "chr17\t41196312\t41277500\tBRCA1\n",
            "chr13\t32889617\t32973809\tBRCA2\n",
            "\n",
            "=== Verificar formato (deve ter TABs) ===\n",
            "chr17^I41196312^I41277500^IBRCA1$\n",
            "chr13^I32889617^I32973809^IBRCA2$\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# 1. Verificar qual arquivo BED você está usando\n",
        "!echo \"=== Conteúdo do alvos.bed (ERRADO) ===\"\n",
        "!cat bioinfotest/data/alvos.bed\n",
        "\n",
        "# 2. Criar o arquivo BED CORRETO com BRCA1 e BRCA2\n",
        "with open('bioinfotest/data/BRCA.bed', 'w') as f:\n",
        "    f.write(\"chr17\\t41196312\\t41277500\\n\")  # BRCA1\n",
        "    f.write(\"chr13\\t32889617\\t32973809\\n\")  # BRCA2\n",
        "\n",
        "!echo -e \"\\n=== Conteúdo do BRCA.bed (CORRETO) ===\"\n",
        "!cat bioinfotest/data/BRCA.bed\n",
        "\n",
        "# 3. Verificar se há cobertura nas regiões BRCA\n",
        "BAM = \"bioinfotest/output/10-7-BRCA.sorted.bam\"\n",
        "!echo -e \"\\n=== Cobertura em BRCA1 ===\"\n",
        "!samtools view -c $BAM chr17:41196312-41277500\n",
        "\n",
        "!echo -e \"\\n=== Cobertura em BRCA2 ===\"\n",
        "!samtools view -c $BAM chr13:32889617-32973809\n",
        "\n",
        "# 4. RE-EXECUTAR FreeBayes com o arquivo CORRETO\n",
        "REF = \"bioinfotest/data/hg19.fasta\"\n",
        "BED = \"bioinfotest/data/BRCA.bed\"  # ARQUIVO CORRETO!\n",
        "VCF_OUT = \"bioinfotest/output/10-7-BRCA.raw.vcf\"\n",
        "\n",
        "!freebayes -f $REF --targets $BED $BAM > $VCF_OUT\n",
        "\n",
        "# 5. Verificar quantas variantes foram encontradas\n",
        "!echo -e \"\\n=== Variantes encontradas no VCF raw ===\"\n",
        "!grep -v \"^#\" $VCF_OUT | wc -l\n",
        "!grep -v \"^#\" $VCF_OUT | head -5\n",
        "\n",
        "# 6. Re-anotar com SnpEff\n",
        "VCF_ANN = \"bioinfotest/output/10-7-BRCA.annotated.vcf\"\n",
        "!java -Xmx4g -jar snpEff/snpEff.jar hg19 $VCF_OUT > $VCF_ANN\n",
        "\n",
        "# 7. Verificar variantes anotadas\n",
        "!echo -e \"\\n=== Variantes anotadas ===\"\n",
        "!grep -v \"^#\" $VCF_ANN | wc -l"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "6sYMUkZB_Ax2",
        "outputId": "53f1c227-aead-4b1c-fb0b-e6bb2b4d1df7"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "=== Conteúdo do alvos.bed (ERRADO) ===\n",
            "chr1\t35000\t36000\n",
            "\n",
            "=== Conteúdo do BRCA.bed (CORRETO) ===\n",
            "chr17\t41196312\t41277500\n",
            "chr13\t32889617\t32973809\n",
            "\n",
            "=== Cobertura em BRCA1 ===\n",
            "38743\n",
            "\n",
            "=== Cobertura em BRCA2 ===\n",
            "45659\n",
            "\n",
            "=== Variantes encontradas no VCF raw ===\n",
            "84\n",
            "chr17\t41215292\t.\tT\tC\t2.21765e-13\t.\tAB=0;ABP=0;AC=0;AF=0;AN=2;AO=2;CIGAR=1X;DP=33;DPB=33;DPRA=0;EPP=3.0103;EPPR=3.64073;GTI=0;LEN=1;MEANALT=1;MQM=60;MQMR=60;NS=1;NUMALT=1;ODDS=30.6797;PAIRED=1;PAIREDR=1;PAO=0;PQA=0;PQR=0;PRO=0;QA=74;QR=1123;RO=31;RPL=0;RPP=7.35324;RPPR=70.3259;RPR=2;RUN=1;SAF=1;SAP=3.0103;SAR=1;SRF=17;SRP=3.64073;SRR=14;TYPE=snp\tGT:DP:AD:RO:QR:AO:QA:GL\t0/0:33:31,2:31:1123:2:74:0,-2.90856,-94.342\n",
            "chr17\t41215294\t.\tT\tC\t2.93065e-14\t.\tAB=0;ABP=0;AC=0;AF=0;AN=2;AO=2;CIGAR=1X;DP=35;DPB=35;DPRA=0;EPP=3.0103;EPPR=3.60252;GTI=0;LEN=1;MEANALT=1;MQM=60;MQMR=60;NS=1;NUMALT=1;ODDS=33.7692;PAIRED=1;PAIREDR=1;PAO=0;PQA=0;PQR=0;PRO=0;QA=72;QR=1228;RO=33;RPL=0;RPP=7.35324;RPPR=74.6689;RPR=2;RUN=1;SAF=1;SAP=3.0103;SAR=1;SRF=18;SRP=3.60252;SRR=15;TYPE=snp\tGT:DP:AD:RO:QR:AO:QA:GL\t0/0:35:33,2:33:1228:2:72:0,-3.70014,-103.981\n",
            "chr17\t41215295\t.\tT\tC\t3.39026e-14\t.\tAB=0;ABP=0;AC=0;AF=0;AN=2;AO=2;CIGAR=1X;DP=35;DPB=35;DPRA=0;EPP=3.0103;EPPR=3.60252;GTI=0;LEN=1;MEANALT=1;MQM=60;MQMR=60;NS=1;NUMALT=1;ODDS=32.8971;PAIRED=1;PAIREDR=1;PAO=0;PQA=0;PQR=0;PRO=0;QA=76;QR=1246;RO=33;RPL=0;RPP=7.35324;RPPR=74.6689;RPR=2;RUN=1;SAF=1;SAP=3.0103;SAR=1;SRF=18;SRP=3.60252;SRR=15;TYPE=snp\tGT:DP:AD:RO:QR:AO:QA:GL\t0/0:35:33,2:33:1246:2:76:0,-3.32138,-105.222\n",
            "chr17\t41215813\t.\tT\tC\t0\t.\tAB=0;ABP=0;AC=0;AF=0;AN=2;AO=4;CIGAR=1X;DP=56;DPB=56;DPRA=0;EPP=3.0103;EPPR=13.7006;GTI=0;LEN=1;MEANALT=1;MQM=60;MQMR=60;NS=1;NUMALT=1;ODDS=43.081;PAIRED=1;PAIREDR=1;PAO=0;PQA=0;PQR=0;PRO=0;QA=146;QR=1951;RO=52;RPL=0;RPP=11.6962;RPPR=115.927;RPR=4;RUN=1;SAF=2;SAP=3.0103;SAR=2;SRF=34;SRP=13.7006;SRR=18;TYPE=snp\tGT:DP:AD:RO:QR:AO:QA:GL\t0/0:56:52,4:52:1951:4:146:0,-3.3611,-162.344\n",
            "chr17\t41222843\t.\tG\tT\t9.90707e-05\t.\tAB=0.111111;ABP=26.6552;AC=1;AF=0.5;AN=2;AO=2;CIGAR=1X;DP=18;DPB=18;DPRA=0;EPP=3.0103;EPPR=16.582;GTI=0;LEN=1;MEANALT=1;MQM=60;MQMR=60;NS=1;NUMALT=1;ODDS=10.6882;PAIRED=1;PAIREDR=1;PAO=0;PQA=0;PQR=0;PRO=0;QA=76;QR=582;RO=16;RPL=0;RPP=7.35324;RPPR=37.7539;RPR=2;RUN=1;SAF=1;SAP=3.0103;SAR=1;SRF=13;SRP=16.582;SRR=3;TYPE=snp\tGT:DP:AD:RO:QR:AO:QA:GL\t0/1:18:16,2:16:582:2:76:-1.79613,0,-47.2949\n",
            "\n",
            "=== Variantes anotadas ===\n",
            "84\n"
          ]
        }
      ]
    }
  ],
  "metadata": {
    "colab": {
      "provenance": [],
      "authorship_tag": "ABX9TyM457OMmGLDcTrts8lxZtt7"
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}